Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/TapToPlace.cs

```diff
@@ -0,0 +1,373 @@
+        private float surfaceNormalOffset = 0.0f;
+
+        /// <summary>
+        /// The distance between the center of the game object to place and a surface along the surface normal, if the raycast hits a surface.
+        /// </summary>
+        public float SurfaceNormalOffset
+        {
+            get => surfaceNormalOffset;
+            set => surfaceNormalOffset = value;
+        }
+
+        [SerializeField]
+        [Tooltip("If true, the game object to place will remain upright and in line with Vector3.up")]
+        private bool keepOrientationVertical = false;
+
+        /// <summary>
+        /// If true, the game object to place will remain upright and in line with Vector3.up
+        /// </summary>
+        public bool KeepOrientationVertical
+        {
+            get => keepOrientationVertical;
+            set => keepOrientationVertical = value;
+        }
+
+        [SerializeField]
+        [Tooltip("If false, the game object to place will not change its rotation according to the surface hit.  The object will" +
+            " remain facing the camera while IsBeingPlaced is true.  If true, the object will rotate according to the surface normal" +
+            " if there is a hit.")]
+        private bool rotateAccordingToSurface = false;
+
+        /// <summary>
+        /// If false, the game object to place will not change its rotation according to the surface hit.  The object will
+        /// remain facing the camera while IsBeingPlaced is true.  If true, the object will rotate according to the surface normal
+        /// if there is a hit.
+        /// </summary>
+        public bool RotateAccordingToSurface
+        {
+            get => rotateAccordingToSurface;
+            set => rotateAccordingToSurface = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Array of LayerMask to execute from highest to lowest priority. First layermask to provide a raycast hit will be used by component.")]
+        private LayerMask[] magneticSurfaces = { UnityEngine.Physics.DefaultRaycastLayers };
+
+        /// <summary>
+        /// Array of LayerMask to execute from highest to lowest priority. First layermask to provide a raycast hit will be used by component.
+        /// </summary>
+        public LayerMask[] MagneticSurfaces
+        {
+            get => magneticSurfaces;
+            set => magneticSurfaces = value;
+        }
+
+        [SerializeField]
+        [Tooltip("If true and in the Unity Editor, the normal of the raycast hit will be drawn in yellow.")]
+        private bool debugEnabled = true;
+
+        /// <summary>
+        /// If true and in the Unity Editor, the normal of the raycast hit will be drawn in yellow.
+        /// </summary>
+        public bool DebugEnabled
+        {
+            get => debugEnabled;
+            set => debugEnabled = value;
+        }
+
+        [SerializeField, FormerlySerializedAs("OnPlacingStarted")]
```

Comment:
```suggestion
        [SerializeField]
```

Reply:
Because this is the first time we're checking this in, there's no need to have FormerlySerializedAs. If the issue here is solving a problem in the example Unity scene (i.e. it still has OnPlacingStarted) we should just be able to reserialize the .unity scene and make it update to this new hotness.

Reply:
Removed